cirq-core/cirq/circuits/circuit.py

```diff
@@ -573,14 +573,12 @@ def enqueue_next(qubit: 'cirq.Qid', moment: int) -> None:
             continue_past = (
                 cur_op is not None and active.issuperset(cur_op.qubits) and not is_blocker(cur_op)
             )
-            if continue_past:
-                for q in cur_op.qubits:
+            for q in cur_op.qubits:
+                if continue_past:
                     enqueue_next(q, cur_moment + 1)
-            else:
-                for q in cur_op.qubits:
-                    if q in active:
-                        end_frontier[q] = cur_moment
-                        active.remove(q)
+                elif q in active:
+                    end_frontier[q] = cur_moment
+                    active.remove(q)
```

Comment:
This is shorter, but tests the condition on each iteration. I think the previous code is slightly better.

Reply:
makes sense to me! pushed.